src/test/java/uk/ac/ebi/subs/ena/EnaReadSubmission.java

```diff
@@ -0,0 +1,154 @@
+        token = TestUtils.getJWTToken(pm.getAuthenticationUrl(), pm.getAapUsername(), pm.getAapPassword());
+        submissionUrl = TestUtils.createSubmission(token, pm.getSubmissionsApiTemplatedUrl(), pm.getSubmitterEmail(), pm.getTeamName());
+        TestUtils.createProject(token, pm.getProjectsApiBaseUrl(), submissionUrl, projectAlias);
+    }
+
+    @Test
+    public void A_addStudy() throws Exception {
+        String studyUrl = TestUtils.createStudy(token, pm.getStudiesApiBaseUrl(), submissionUrl, studyAlias, projectAlias, pm.getTeamName());
+        TestUtils.waitForValidationResults(token, studyUrl);
+        ValidationResult validationResult = TestUtils.getValidationResultForSubmittable(studyUrl, token);
+        assertNoErrorsInValidationResult(validationResult);
+    }
+
+    @Test
+    public void B_addSample() throws Exception {
+        String sampleJson = TestJsonUtils.getSampleJson(submissionUrl, sampleAlias);
+        String sampleUrl = TestUtils.createSubmittable(token, pm.getSamplesApiBaseUrl(), sampleJson);
+        TestUtils.waitForValidationResults(token, sampleUrl);
+        ValidationResult validationResult = TestUtils.getValidationResultForSubmittable(sampleUrl, token);
+        assertNoErrorsInValidationResult(validationResult);
+    }
+
+    @Test
+    public void C_addAssay() throws Exception {
+        String assayUrl = TestUtils.createAssay(token, pm.getAssaysApiBaseUrl(), submissionUrl, assayAlias, studyAlias, sampleAlias);
+        TestUtils.waitForValidationResults(token, assayUrl);
+        ValidationResult validationResult = TestUtils.getValidationResultForSubmittable(assayUrl, token);
+        assertNoErrorsInValidationResult(validationResult);
+    }
+
+    @Test
+    public void D_uploadFile() throws Exception{
+        File testFile = new File(ClassLoader.getSystemClassLoader().getResource(fileName).getFile());
+        HttpResponse apiRootResponse = HttpUtils.httpGet(token,pm.getApiRoot());
+        ApiRoot apiRoot = HttpUtils.retrieveResourceFromResponse(apiRootResponse,ApiRoot.class);
+
+        String submissionUUID = submissionUrl.substring(submissionUrl.lastIndexOf('/') + 1);
+        Uploader.uploadFile(token, apiRoot.get_links().getTusUpload().getHref(), testFile, submissionUUID, fileName);
+    }
+
+
+    @Test
+    public void E_addAssayData() throws Exception {
+        String assayDataUrl = TestUtils.createAssayData(
+                token,
+                pm.getAssayDataApiBaseUrl(),
+                submissionUrl,
+                assayDataAlias,
+                assayAlias,
+                fileName,
+                fileType
+        );
+        TestUtils.waitForValidationResults(token, assayDataUrl);
+        ValidationResult validationResult = TestUtils.getValidationResultForSubmittable(assayDataUrl, token);
+        assertNoErrorsInValidationResult(validationResult); //FIXME - unexpected errors from the ENA validator, probably need to fix the validator
+    }
```

Comment:
Up to this point we don't check the validation result of the uploaded file.
I think we should check that one, as well.

Reply:
I've added a check for that, thanks for the pointer